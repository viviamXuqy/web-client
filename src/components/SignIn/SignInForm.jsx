import React, { PureComponent } from 'react';
import { object, func } from 'prop-types';
import { Form, Input, Checkbox, Button } from 'antd';
import { injectIntl } from 'react-intl';

class SignIn extends PureComponent {
  static propTypes = {
    intl: object.isRequired,
    form: object.isRequired,
    signIn: func.isRequired,
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.form.validateFields((err, values) => {
      if (!err) {
        this.props.signIn(values);
      }
    });
  }

  render() {
    const { intl: { formatMessage } } = this.props;
    const { getFieldDecorator } = this.props.form;

    return (
      <Form
        layout="vertical"
        onSubmit={this.handleSubmit}
      >
        <Form.Item label={formatMessage({ id: 'App.signIn.user.label' })}>
          {getFieldDecorator('username', {
            rules: [
              {
                required: true,
                message: formatMessage({ id: 'App.signIn.user.msg' }),
              },
            ],
          })(<Input placeholder={formatMessage({ id: 'App.signIn.user.msg' })} />)}
        </Form.Item>
        <Form.Item label={formatMessage({ id: 'App.signIn.pwd.label' })}>
          {getFieldDecorator('password', {
            rules: [
              {
                required: true,
                message: formatMessage({ id: 'App.signIn.pwd.msg' }),
              },
            ],
          })(<Input type="password" placeholder={formatMessage({ id: 'App.signIn.pwd.msg' })} />)}
        </Form.Item>
        <Form.Item className="form__footer">
          {getFieldDecorator('remember', {
            valuePropName: 'checked',
          })(<Checkbox size="large" className="form__remember">{formatMessage({ id: 'App.signIn.remember' })}</Checkbox>)}
          <Button size="large" type="primary" htmlType="submit" className="form__btn">
            {formatMessage({ id: 'App.signIn.button' })}
          </Button>
        </Form.Item>
      </Form>
    );
  }
}

export default Form.create()(injectIntl(SignIn));
