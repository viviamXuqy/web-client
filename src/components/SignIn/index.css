a {
  text-decoration: none;
}

.sign-page {
  padding: 0;
  background-image: url("../../assets/images/login_bg.jpg");
  background-size: cover;
  background-position: 50%;
  background-repeat: no-repeat;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.sign-page:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  /* background-color: rgba(0,0,0,.8); */
}

.sign-page__box {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  display: flex;
  width: 900px;
  background-color: #fff;
  justify-content: space-around;
  box-shadow: 0 2px 10px rgba(0,0,0,.3);
}

.sign-page__box > div {
  width: 50%;
}

.sign-page__slogan {
  position: relative;
  padding: 50px;
  background-color: #153F71;
}

.sign-page__slogan > h1 {
  font-size: 72px;
  line-height: 1.2;
  color: #fff;
}

.sign-page__slogan > h2 {
  font-size: 24px;
  color: #fff;
}

.sign-page__slogan .app-logo {
  position: absolute;
  left: 50px;
  bottom: 50px;
}

.sign-page__slogan .app-logo__title {
  color: #fff;
}

.sign-page__form {
  width: 50%;
  padding: 50px;
}

.sign-page__form h1 {
  font-size: 56px;
  margin-bottom: 20px;
}

.sign-page__form .ant-form-item-label label {
  font-size: 16px;
}

.sign-page__form .ant-form-item-required:before {
  content: '';
}

.sign-page__form .form__footer {
  margin: 60px 0 0;
}

.form__remember {
  margin-top: 4px;
}

.form__btn {
  width: 50%;
  float: right;
  border-radius: 100px;
  height: 40px;
}