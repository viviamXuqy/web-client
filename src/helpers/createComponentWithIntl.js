import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line
import LocaleProvider from 'containers/Locale';
import { setStore } from '../utils/storage';
import { nodeWithIntlProp } from './intl-enzyme-test-helper';

const createComponentWithIntl = (children, props = { lang: 'zh-CN' }) => {
  setStore('locale', props.lang);
  return renderer.create(
    <LocaleProvider {...props}>
      {nodeWithIntlProp(children)}
    </LocaleProvider>,
  );
};

export default createComponentWithIntl;
