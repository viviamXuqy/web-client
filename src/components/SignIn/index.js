import React from 'react';
import { object, func } from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';

import { getLocalLogo } from '../../utils/locale';
import SignInForm from './SignInForm';
import './index.css';


const SignIn = ({
  intl,
  signIn,
}) => (
  <div className="main-wrapper sign-page">
    <div className="sign-page__box">
      <div className="sign-page__slogan">
        <h1>
          Hello,<br />
          Welcome<br />
          BackÔºÅ
        </h1>
        <h2><FormattedMessage id="App.signIn.info" /></h2>
        <div className="app-logo">
          <img src={getLocalLogo(intl.locale)} alt="Logo" />
        </div>
      </div>
      <div className="sign-page__form">
        <h1><FormattedMessage id="App.signIn.title" /></h1>
        <SignInForm signIn={signIn} />
      </div>
    </div>
  </div>
);

SignIn.propTypes = {
  intl: object.isRequired,
  signIn: func.isRequired,
};

export default injectIntl(SignIn);
