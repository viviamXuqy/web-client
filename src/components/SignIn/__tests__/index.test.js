import React from 'react';
import createComponentWithIntl from 'helpers/createComponentWithIntl';
import { mountWithIntl } from 'helpers/intl-enzyme-test-helper';

import SignIn from 'components/SignIn';

const initProps = {
  signIn: () => {},
  lang: 'zh-CN',
};
const setup = (props = initProps) => {
  const component = <SignIn {...props} />;
  const componentWithProvider = createComponentWithIntl(component);
  const wrapper = mountWithIntl(component);

  return {
    componentWithProvider,
    wrapper,
  };
};

describe('SignIn component', () => {
  it('should render without errors', () => {
    const { wrapper } = setup();

    expect(wrapper).toBeTruthy();
  });

  it('should render correct things', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.sign-page').length).toBe(1);
  });

  it("should render right thing and don't change unexpected", () => {
    const { componentWithProvider } = setup();
    const tree = componentWithProvider.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
