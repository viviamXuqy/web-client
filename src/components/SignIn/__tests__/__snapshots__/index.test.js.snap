// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render right thing and don't change unexpected 1`] = `
<div
  className="main-wrapper sign-page"
>
  <div
    className="sign-page__box"
  >
    <div
      className="sign-page__slogan"
    >
      <h1>
        Hello,
        <br />
        Welcome
        <br />
        Back！
      </h1>
      <h2>
        <span>
          百眼卫士视觉信息结构化平台
        </span>
      </h2>
      <div
        className="app-logo"
      >
        <img
          alt="Logo"
          src="logo.svg"
        />
      </div>
    </div>
    <div
      className="sign-page__form"
    >
      <h1>
        <span>
          登录
        </span>
      </h1>
      <form
        className="ant-form ant-form-vertical"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={undefined}
        >
          <div
            className="ant-form-item-label"
          >
            <label
              className="ant-form-item-required"
              htmlFor="username"
              onClick={[Function]}
              title="用户名"
            >
              用户名
            </label>
          </div>
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  data-__field={
                    Object {
                      "name": "username",
                    }
                  }
                  data-__meta={
                    Object {
                      "name": "username",
                      "originalProps": Object {
                        "disabled": false,
                        "prefixCls": "ant-input",
                        "type": "text",
                      },
                      "ref": null,
                      "rules": Array [
                        Object {
                          "message": "请输入用户名",
                          "required": true,
                        },
                      ],
                      "trigger": "onChange",
                      "validate": Array [
                        Object {
                          "rules": Array [
                            Object {
                              "message": "请输入用户名",
                              "required": true,
                            },
                          ],
                          "trigger": Array [
                            "onChange",
                          ],
                        },
                      ],
                      "valuePropName": "value",
                    }
                  }
                  disabled={false}
                  id="username"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="text"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={undefined}
        >
          <div
            className="ant-form-item-label"
          >
            <label
              className="ant-form-item-required"
              htmlFor="password"
              onClick={[Function]}
              title="密码"
            >
              密码
            </label>
          </div>
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <input
                  className="ant-input"
                  data-__field={
                    Object {
                      "name": "password",
                    }
                  }
                  data-__meta={
                    Object {
                      "name": "password",
                      "originalProps": Object {
                        "disabled": false,
                        "prefixCls": "ant-input",
                        "type": "password",
                      },
                      "ref": null,
                      "rules": Array [
                        Object {
                          "message": "请输入密码",
                          "required": true,
                        },
                      ],
                      "trigger": "onChange",
                      "validate": Array [
                        Object {
                          "rules": Array [
                            Object {
                              "message": "请输入密码",
                              "required": true,
                            },
                          ],
                          "trigger": Array [
                            "onChange",
                          ],
                        },
                      ],
                      "valuePropName": "value",
                    }
                  }
                  disabled={false}
                  id="password"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  type="password"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item form__footer"
          style={undefined}
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <label
                  className="form__remember ant-checkbox-wrapper"
                  onMouseEnter={undefined}
                  onMouseLeave={undefined}
                  style={undefined}
                >
                  <span
                    className="ant-checkbox"
                    style={Object {}}
                  >
                    <input
                      autoFocus={undefined}
                      checked={false}
                      className="ant-checkbox-input"
                      data-__field={
                        Object {
                          "name": "remember",
                        }
                      }
                      data-__meta={
                        Object {
                          "name": "remember",
                          "originalProps": Object {
                            "children": "记住登录状态",
                            "className": "form__remember",
                            "indeterminate": false,
                            "prefixCls": "ant-checkbox",
                            "size": "large",
                          },
                          "ref": null,
                          "trigger": "onChange",
                          "validate": Array [],
                          "valuePropName": "checked",
                        }
                      }
                      disabled={undefined}
                      id="remember"
                      name={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={undefined}
                      onFocus={[Function]}
                      readOnly={undefined}
                      tabIndex={undefined}
                      type="checkbox"
                      value={undefined}
                    />
                    <span
                      className="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    记住登录状态
                  </span>
                </label>
                <button
                  className="ant-btn form__btn ant-btn-primary ant-btn-lg"
                  onClick={[Function]}
                  type="submit"
                >
                  <span>
                    登 录
                  </span>
                </button>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
